hazelcast/src/main/java/com/hazelcast/internal/server/ServerConnectionManager.java

```diff
@@ -213,4 +215,18 @@ default boolean transmit(Packet packet, Address target) {
      * @return the Server.
      */
     Server getServer();
+
+    /**
+     * blocks the caller thread until a connection is established (or failed)
+     * or the time runs out
```

Comment:
```suggestion
     * or the time runs out. Callers must ensure a connection is established after this method returns {@code true}.
```
Reasoning: we are replacing `sleep`s with `blockOnConnect` and these checks are already in place. In the implementation, we return `true` if no entry is found in `connectionsInProgress` map but I think that this does not guarantee that the connection was established, therefore a further check on the caller side is required. wdyt?